### PR TITLE
Setup assosiations

### PR DESCRIPTION
**Pull request by hiromataba**
Int his PR , I did the following :point_down: 
- Create a class Classroom with the following:
    - @label instance variable, should be initialized in the constructor.
    - Setter and getter for @label (remember about attr_accessor).
    
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
    - Create the has-many side (a classroom has many students).
    - Create the belongs-to side (a student belongs to a classroom).
   - Make sure that when adding a student to a classroom it also sets the classroom for the student.
   - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
   
- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
  
- Create a class Rental with the following:
   - @date instance variable, should be initialized in the constructor.
   - Setter and getter for @date (remember about attr_accessor).
   
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
   - Create the has-many side of Book and Rental (a book has many rentals).
   - Create the belongs-to side of Rental and Book (a rental belongs to a book).
   - Create the has-many side of Person and Rental (a person has many rentals).
   - Create the belongs-to side of Rental and Person (a rental belongs to a person).
   - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (AshakaE) [inline book.rb]: we need to make rentals read/write, right now it's set to only read

- [ ] Please update the setter for rentals

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (AshakaE) [inline book.rb]: we need a method for adding rentals for a book, see [reference](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

- [ ] Kindly add a method for creating rentals

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (AshakaE) [inline rental.rb]: we need to make book and person read/write, right now it's set to only read

- [ ] Please update the setter for book and person

File: person.rb [hunk 4]
```diff
@@ -2,13 +2,14 @@
 
 class Person
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (AshakaE) [inline person.rb]: we  also need to make rentals read/write, right now it's set to only read

- [ ] Please update the setter for rentals

File: person.rb [hunk 5]
```diff
@@ -2,13 +2,14 @@
...
 
   def initialize(age:, name: 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     @corrector = Corrector.new
   end
```
Comment (AshakaE) [inline person.rb]: A person should be able to make rent a book

we need a method for adding rentals for a book, see [reference](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

- [ ] Kindly add a method for creating rentals

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @hiromataba  🙋‍♀️, 

Good job so far! 👌

Git flow ✔️ 
Pull request is 👌 
Readme ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Comment by hiromataba** [comment]
Thank you for the feedback :+1:

File: person.rb (hunk 5 above)
Comment (hiromataba) [reply person.rb]: Thank you for the feedback :pray:

**Pull request merged**